<launch>
  <arg name="config_path" default = "/home/heyijia/ros_ws/calibra_laser_cam_ws/src/config_and_tool/calibrat_config.yaml" />


  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
    <!-- Remap topic required by the node to custom topics -->
    <!--<remap from="image_rect" to="/mynteye/left/image_raw" />-->

    <param name="config_file" type="string" value="$(arg config_path)" />

    <!-- Optional: Subscribe to the compressed stream-->
    <param name="image_transport" type="str" value="compressed" />

    <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
    <param name="tag_family" type="str" value="36h11" />

    <!-- Enable projected optical measurements for more accurate tag transformations -->
    <!-- This exists for backwards compatability and should be left true for new setups -->
    <param name="projected_optics" type="bool" value="true" />

    <!-- Describe the tags -->
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.1612},
      {id: 1, size: 0.1612, frame_id: a_frame},
      {id: 2, size: 0.1612, frame_id: tag_2},
      {id: 3, size: 0.1612},
      {id: 4, size: 0.1612},
      {id: 5, size: 0.1612},
      {id: 6, size: 0.1612},
      {id: 7, size: 0.1612},
      {id: 8, size: 0.1612},
      {id: 9, size: 0.1612},
      {id: 10, size: 0.1612}]
    </rosparam>
  </node>
</launch>
